### PR TITLE
[Exporter.OpenTelemetryProtocol] User agent changed to OTel-OTLP-Exporter-Dotnet/{NuGet Package Version}

### PR DESCRIPTION
**Pull request by Kielek**
Follow up to #5498

## Changes

Handles following comments:

* https://github.com/open-telemetry/opentelemetry-dotnet/pull/5498#discussion_r1554197469 (User-Agent part, other was handled previously)
* https://github.com/open-telemetry/opentelemetry-dotnet/pull/5498#discussion_r1558073568
* https://github.com/open-telemetry/opentelemetry-dotnet/pull/5498#discussion_r1558076614

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* ~~[ ] Unit tests added/updated~~
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* ~~[ ] Changes in public API reviewed (if applicable)~~

### CODE REVIEW

**Comment by Kielek** [comment]
@CodeBlanch, @reyang, FYI

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,11 @@
 * Fix native AoT warnings in `OpenTelemetry.Exporter.OpenTelemetryProtocol`.
   ([#5520](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5520))
 
+* `User-Agent` header changed from
+  `OTel-OTLP-Exporter-Dotnet-{NuGet Package Version}+{Commit Hash}`
+  to `OTel-OTLP-Exporter-Dotnet-{NuGet Package Version}`.
```
Comment (Kielek) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md:10]:
```suggestion
  `OTel-OTLP-Exporter-Dotnet/{NuGet Package Version}+{Commit Hash}`
  to `OTel-OTLP-Exporter-Dotnet/{NuGet Package Version}`.
```

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs [hunk 2]
```diff
@@ -229,9 +228,8 @@ private static string GetUserAgentString()
     {
         try
         {
-            var assemblyVersion = typeof(OtlpExporterOptions).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>();
-            var informationalVersion = assemblyVersion?.InformationalVersion;
-            return string.IsNullOrEmpty(informationalVersion) ? UserAgentProduct : $"{UserAgentProduct}/{informationalVersion}";
+            var assembly = typeof(OtlpExporterOptions).Assembly;
+            return $"{UserAgentProduct}/{assembly.GetPackageVersion()}";
```
Comment (reyang) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs:232]: Minor, remove `UserAgentProduct` and update this line to:

```suggestion
            return $"OTel-OTLP-Exporter-Dotnet/{assembly.GetPackageVersion()}";
```

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,11 @@
 * Fix native AoT warnings in `OpenTelemetry.Exporter.OpenTelemetryProtocol`.
   ([#5520](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5520))
 
+* `User-Agent` header changed from
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md:8]:
```suggestion
* `User-Agent` header format changed from
```

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs (hunk 2 above)
Comment (Kielek) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptions.cs]: b45be344b341745c895b3eaf0bd61e1303a0cf30

**Review by CodeBlanch — APPROVED** [review]
LGTM

**Pull request merged**